src/pathTemplate.ts

```diff
@@ -66,45 +46,46 @@ export class PathTemplate {
    * @throws {TypeError} if path can't be matched to this template
    */
   match(path: string): Bindings {
-    const pathSegments = path.split('/');
+    let pathSegments = path.split('/');
+
     const bindings: Bindings = {};
-    let segmentCount = this.size;
-    let current: string;
-    let index = 0;
-    this.segments.forEach(segment => {
-      if (index > pathSegments.length) {
-        return;
+    if (pathSegments.length !== this.segments.length) {
+      // if the path contains a wildcard, then the length may differ by 1.
+      if (!this.data.includes('**')) {
+        throw new TypeError(
+          `This path ${path} does not match, the parameter number is not same.`
```

Comment:
This path does not match what exactly? ("match" is something that has 2 parameters, "a matches b", and it makes sense to have both in the error message)

Reply:
reasonable, i have include `this.data` (original path template)  in the error messages.